workflows/src/main/kotlin/com/r3/corda/lib/accounts/workflows/flows/RequestAccountInfoByNameFlows.kt

```diff
@@ -0,0 +1,74 @@
+package com.r3.corda.lib.accounts.workflows.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.corda.lib.accounts.contracts.states.AccountInfo
+import com.r3.corda.lib.accounts.workflows.internal.accountService
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.node.StatesToRecord
+import net.corda.core.utilities.unwrap
+
+/**
+ * This flow can be used to check whether an account is hosted by a particular host by passing account name. If it is then the host will share
+ * the account info with the requester. This flow should be used in a situation where UUID is difficult to obtain for a given
+ * account. For e.g: a node can create accounts for its employees using SSN/NIN or employeeId
+ *
+ * @property accountName account name to request the [AccountInfo] for hosted at [host] node.
+ * @property host session to request the [AccountInfo] from.
+ */
+class RequestAccountInfoByNameFlow(private val accountName: String, val host: FlowSession) : FlowLogic<AccountInfo?>() {
+    @Suspendable
+    override fun call(): AccountInfo? {
+        val hasAccount = host.sendAndReceive<Boolean>(accountName).unwrap { it }
+        return if (hasAccount) subFlow(ShareAccountInfoHandlerFlow(host)) else null
+    }
+}
+
+/**
+ * Responder flow for [RequestAccountInfoByNameFlow].
+ */
+class RequestAccountInfoByNameHandlerFlow(val otherSession: FlowSession) : FlowLogic<Unit>() {
+    @Suspendable
+    override fun call() {
+        val accountName = otherSession.receive(String::class.java).unwrap {it }
+        val response = serviceHub.accountService.accountInfo(accountName).find { it.state.data.host == ourIdentity }
+        if (response == null) {
+            otherSession.send(false)
+        } else {
+            otherSession.send(true)
+            subFlow(ShareAccountInfoFlow(response, listOf(otherSession)))
+        }
+    }
+}
+
+// Initiating versions of the above flows.
+
+/**
+ * Shares an [AccountInfo] [StateAndRef] with the supplied [Party]s. The [AccountInfo] is always stored using
+ * [StatesToRecord.ALL_VISIBLE].
+ *
+ * @property accountName account name to request the [AccountInfo] for.
+ * @property host [Party] to request the [AccountInfo] from.
+ */
+@StartableByRPC
+@StartableByService
+@InitiatingFlow
+class RequestAccountInfoByName(private val accountName: String, val host: Party) : FlowLogic<AccountInfo?>() {
+    @Suspendable
+    override fun call(): AccountInfo? {

```

Comment:
If `ourIdentity == host`, then you don't need to continue; just fetch the account from the current node.

Reply:
Thanks for the comments.
This flow is specifically to request accountInfo from another node and not from the caller's local node itself. The assumption here is that the caller is aware that an account with name "charlie" exists on host node "Alice". (obviously the caller will try to search for account with name "Charlie" on its local vault first)

Reply:
In my humble opinion, you cannot assume how your flow will be used; your code should take care of all cases.   
For instance in the `RequestKeyForAccountFlow` (which comes with the library):  

1. The flow returns the key immediately if the `host` is the initiator (see [here](https://github.com/corda/accounts/blob/900fe9c2ebf4ad8ecda1f494c03f747ef1579746/workflows/src/main/kotlin/com/r3/corda/lib/accounts/workflows/flows/RequestKeyForAccountFlows.kt#L35))   
2. Otherwise, it contacts the `host` node (see [here](https://github.com/corda/accounts/blob/900fe9c2ebf4ad8ecda1f494c03f747ef1579746/workflows/src/main/kotlin/com/r3/corda/lib/accounts/workflows/flows/RequestKeyForAccountFlows.kt#L38))

Reply:
Well, the design of this flow is kept in consistent with the design of RequestAccountInfoFlow ([here](https://github.com/corda/accounts/blob/master/workflows/src/main/kotlin/com/r3/corda/lib/accounts/workflows/flows/RequestAccountInfoFlows.kt))

Reply:
Ah, I see; now I'm curious why they did it that way 🙂